src/project.c

```diff
@@ -1346,3 +1346,25 @@ void project_init(void)
 void project_finalize(void)
 {
 }
+
+
+/**
+ * Open the given project file.
+ *
+ * This is done as if the user had done it using the `Project->Open`
+ * menu item (ie. it will confirm closing the current project). Any
+ * files in the project's session will be re-opened.
+ *
+ * @param project_locale_fn The filename of the project file in locale
+ * filename encoding.
+ *
+ * @return @c TRUE if the project was opened, @c FALSE otherwise.
+ *
+ * @since 1.29 (API 230)
+ */
+GEANY_API_SYMBOL
+gboolean project_open_file(const gchar *project_locale_fn)
+{
+	return (project_ask_close() &&
+		project_load_file_with_session(project_locale_fn));
```

Comment:
> I don't think that check is correct though. On Windows it should use `g_utf8_casefold()` because filenames are not case sensitive

Yeah probably.

>  and on all platforms it should probably use `g_utf8_collate_key_for_filename()` or similar to do a closer to "proper" comparison.

No, on non-Windows, filenames are byte sequences generally; and anyway `g_utf8_collate_key_for_filename()` is meant for sorting filenames naturally (e.g. case insensitive, human-readbale numbers, etc.), not compare for path equality.